tests/test_basics.py

```diff
@@ -89,20 +78,35 @@ def error_processor(event, exc_info):
     assert event["exception"]["values"][0]["value"] == "aha! whatever"
 
 
+class ModuleImportErrorSimulator:
+    def __init__(self, modules, error_cls=DidNotEnable):
+        self.modules = modules
+        self.error_cls = error_cls
+        for sys_module in list(sys.modules.keys()):
+            if any(sys_module.startswith(module) for module in modules):
+                del sys.modules[sys_module]
+
+    def find_spec(self, fullname, _path, _target=None):
+        if fullname in self.modules:
+            raise self.error_cls("Test import failure for %s" % fullname)
+
+    def __enter__(self):
+        # WARNING: We need to be first to avoid pytest messing with local imports
+        sys.meta_path.insert(0, self)
+
+    def __exit__(self, *_args):
+        sys.meta_path.remove(self)
+
+
 def test_auto_enabling_integrations_catches_import_error(sentry_init, caplog):
     caplog.set_level(logging.DEBUG)
-    redis_index = _AUTO_ENABLING_INTEGRATIONS.index(
-        "sentry_sdk.integrations.redis.RedisIntegration"
-    )  # noqa: N806
 
-    sentry_init(auto_enabling_integrations=True, debug=True)
+    with ModuleImportErrorSimulator(
+        [i.rsplit(".", 1)[0] for i in _AUTO_ENABLING_INTEGRATIONS]
+    ):
+        sentry_init(auto_enabling_integrations=True, debug=True)
```

Comment:
Afaict this is now testing something slightly different -- before, we were testing that if we get an `ImportError` while trying to import an external library, we throw a `DidNotEnable` and the integration doesn't get enabled, but now we're skipping the `ImportError` -> `DidNotEnable` part and going straight to "if an integration raises `DidNotEnable`, it won't be enabled".

If we change the import error simulator here to throw an `ImportError` instead and run it with a handful of external modules (`django`, `flask`?) instead of the integrations directly, that should then behave as before I think. (We anyway check for _at least one_ import failure at the end, so might not be necessary to turn off all auto-enabling integrations, although who knows, since this test was failing for you with only loguru and pymongo installed. 🙃)

Reply:
I agree and have thought about this. The main issue there is getting a list of integration dependencies reliably and in a maintainable way. 

I'd argue each integration should check its own import issue handling in their separate respective tests as that may not be uniform as we see in the Redis case.

Long story short, my argument is: this patch delivers value as it is and we can address the larger problem I describe above in a follow up.

WDYT?

Reply:
Discussed this elsewhere -- we'll merge this and then look at how we can further improve things later.